src/agent/rustjail/src/container.rs

```diff
@@ -604,9 +605,11 @@ fn do_init_child(cwfd: RawFd) -> Result<()> {
     // Without NoNewPrivileges, we need to set seccomp
     // before dropping capabilities because the calling thread
     // must have the CAP_SYS_ADMIN.
-    if linux.seccomp.is_some() && !oci_process.no_new_privileges {
-        let scmp = linux.seccomp.as_ref().unwrap();
-        seccomp::init_seccomp(scmp)?;
+    #[cfg(feature = "seccomp")]
+    if !oci_process.no_new_privileges {
+        if let Some(ref scmp) = linux.seccomp {
+            seccomp::init_seccomp(scmp)?;
```

Comment:
Shoud we implement the errror handling like below?

```Rust
 if !oci_process.no_new_privileges {
        if let Some(ref scmp) = linux.seccomp {
            #[cfg(feature = "seccomp")]
            seccomp::init_seccomp(scmp)?;
            #[cfg(not(feature = "seccomp"))]
            Err(anyhow!("Seccomp config provided but seccomp not supported"))  
        }
}
```
This error tells users that the runtime enables seccomp support, but the agent does not enable seccomp.

Reply:
This error is handled in the runtime by the [commit](https://github.com/kata-containers/kata-containers/pull/1788/commits/3a41d575243631689e34c3862bc0def61ca9cad9#diff-6fb7f551035031ed29ad100c61f1a7e92b0fa15d2f67e4c7ea7ebf3b715583f4R1371).

Reply:
I think this approach is safer: if the admin asked for seccomp but it isn't available, I'd prefer to get a hard error telling me that immediately. The current code doesn't error, which may lead an admin to _think_ seccomp is being enforced when it isn't (which would be surprising and potentially dangerous).